File: c/Strategy/QuackBehavior.c
#include "QuackBehavior.h"

void QuaBehNew(QuackBehavior *quackbeh, quackFun quack_fun)
{
    quackbeh->quack = quack_fun;
}

void Quack(QuackBehavior *quackbeh)
{
    printf("Quack\n");
}

void MuteQuack(QuackBehavior *quackbeh)
{
    printf("<< Slience >>\n");
}

void SQuack(QuackBehavior *quackbeh)
{
    printf("Squack\n");
}

File: python/Strategy/Duck/QuackBehavior.py
from abc import ABC, abstractmethod


class QuackBehavior(ABC):
    @abstractmethod
    def quack(self):
        pass


class Quack(QuackBehavior):
    def quack(self):
        print("Quack")


class MuteQuack(QuackBehavior):
    def quack(self):
        print("<< Slience >>")


class SQuack(QuackBehavior):
    def quack(self):
        print("Squack")
